### PR TITLE
addapt barman privileges validation Closes #318

### LINKED ISSUE #318: barman check <server> doesn't check access to backup functions correctly
Barman v2.12
PostgreSQL 13.1

Following configuration for server:
```
[ssh-pg-master]
description =  "Master PostgreSQL Database (via ssh)"
ssh_command = ssh postgres@<ip-address>
conninfo = host=<ip-address> user=barman dbname=postgres

backup_method = rsync
backup_options = exclusive_backup
reuse_backup = link
archiver = on

streaming_archiver = on
slot_name = barman
```

User barman is a regular user with the required set of privileges.

Run following command:
`barman --log-level DEBUG check ssh-pg-master`

Result:
Server ssh-pg-master:
    PostgreSQL: OK
    no access to backup functions: FAILED (privileges for PostgreSQL backup functions are required (see documentation))
    PostgreSQL streaming: OK
    wal_level: OK
    replication slot: OK
    directories: OK
    retention policy settings: OK
    backup maximum age: OK (no last_backup_maximum_age provided)
    compression settings: OK
    failed backups: OK (there are 0 failed backups)
    minimum redundancy requirements: OK (have 0 backups, expected at least 0)
    ssh: OK (PostgreSQL server)
    not in recovery: OK
    systemid coherence: OK (no system Id stored on disk)
    pg_receivexlog: OK
    pg_receivexlog compatible: OK
    receive-wal running: OK
    archive_mode: OK
    archive_command: OK
    continuous archiving: OK
    archiver errors: OK

Added barman logfile snippet and postgres lofile snippet as attachments. 

From barman logfile I see:
2021-01-20 17:07:01,427 [9898] barman.server DEBUG: Starting check: 'superuser or standard user with backup privileges'
2021-01-20 17:07:01,427 [9898] barman.server ERROR: Check 'no access to backup functions' failed for server 'ssh-pg-master'

I also see this error:
2021-01-20 17:07:01,280 [9898] barman.postgres DEBUG: Error retrieving PostgreSQL setting 'wal_keep_segments': unrecognized configuration parameter "wal_keep_segments"

wal_keep_segments has become wal_keep_size, but following issue already exists for this one #302 

From postgres logfile I can see that following query is used to check superuser or standard user with backup privileges:
```
SELECT
	          usesuper
	          OR
	          (
	            userepl
	            AND
	            (
	              pg_has_role(CURRENT_USER, 'pg_monitor', 'MEMBER')
	              OR
	              (
	                pg_has_role(CURRENT_USER, 'pg_read_all_settings', 'MEMBER')
	                AND pg_has_role(CURRENT_USER, 'pg_read_all_stats', 'MEMBER')
	              )
	            )
	            AND has_function_privilege(
	              CURRENT_USER, 'pg_start_backup(text,bool,bool)', 'EXECUTE')
	            AND
	            (
	              has_function_privilege(
	                CURRENT_USER, 'pg_stop_backup()', 'EXECUTE')
	              OR has_function_privilege(
	                CURRENT_USER, 'pg_stop_backup(bool,bool)', 'EXECUTE')
	            )
	            AND has_function_privilege(
	              CURRENT_USER, 'pg_switch_wal()', 'EXECUTE')
	            AND has_function_privilege(
	              CURRENT_USER, 'pg_create_restore_point(text)', 'EXECUTE')
	          )
	        FROM
	          pg_user
	        WHERE
	          usename = CURRENT_USER
```
If for the user (in this case barman) attributes superuser and replication (in pg_user usesuper and userpl) are false, then although all privileges for the user are resulting in true, the query will still return false!!

The 'AND' after 'userpl' should be an 'OR' to make the query work correctly:
```
SELECT
	          usesuper
	          OR
	          (
	            userepl
	            OR
	            (
	              pg_has_role(CURRENT_USER, 'pg_monitor', 'MEMBER')
	              OR
	              (
	                pg_has_role(CURRENT_USER, 'pg_read_all_settings', 'MEMBER')
	                AND pg_has_role(CURRENT_USER, 'pg_read_all_stats', 'MEMBER')
	              )
	            )
	            AND has_function_privilege(
	              CURRENT_USER, 'pg_start_backup(text,bool,bool)', 'EXECUTE')
	            AND
	            (
	              has_function_privilege(
	                CURRENT_USER, 'pg_stop_backup()', 'EXECUTE')
	              OR has_function_privilege(
	                CURRENT_USER, 'pg_stop_backup(bool,bool)', 'EXECUTE')
	            )
	            AND has_function_privilege(
	              CURRENT_USER, 'pg_switch_wal()', 'EXECUTE')
	            AND has_function_privilege(
	              CURRENT_USER, 'pg_create_restore_point(text)', 'EXECUTE')
	          )
	        FROM
	          pg_user
	        WHERE
	          usename = CURRENT_USER
```
[barman.log](https://github.com/2ndquadrant-it/barman/files/5844727/barman.log)
[postgres.log](https://github.com/2ndquadrant-it/barman/files/5844728/postgres.log)

### PR DESCRIPTION
**Pull request by didiermichel**

### CODE REVIEW

File: barman/postgres.py [hunk 1]
```diff
@@ -556,7 +556,7 @@ def has_backup_privileges(self):
           OR
           (
             userepl
-            AND
+            OR
```
Comment (amenonsen) [inline barman/postgres.py:559]: For clarity, I think we should rewrite this to say `((userepl OR (…thing…)) AND …otherthings…)`, i.e., put the two OR clauses together inside (). Right now it's just `(userepl OR …thing… AND …otherthings…)`, and I don't think that's right.

File: barman/postgres.py [hunk 2]
```diff
@@ -1325,9 +1325,12 @@ def switch_wal(self):
         """
         try:
             conn = self.connect()
-            # Requires superuser privilege
+            # Requires superuser privilege or specific writes
             if not self.has_backup_privileges:
-                raise BackupFunctionsAccessRequired()
+                msg = "{} user does not have enough privileges to run switch-wal (see documentation).".format(
+                    self.application_name
```
Comment (amenonsen) [inline barman/postgres.py:1331]: This should be the username, not the application name.

File: barman/postgres.py [hunk 3]
```diff
@@ -1385,7 +1388,10 @@ def get_replication_stats(self, client_type=STANDBY):
             cur = self._cursor(cursor_factory=NamedTupleCursor)
 
             if not self.has_backup_privileges:
-                raise BackupFunctionsAccessRequired()
+                msg = "{} user does not have enough privileges to get replication-status (see documentation).".format(
+                    self.application_name
```
Comment (amenonsen) [inline barman/postgres.py:1392]: Ditto.

**Review by amenonsen — CHANGES_REQUESTED** [review]

File: barman/postgres.py (hunk 1 above)
Comment (didiermichel) [reply barman/postgres.py]: Added cosmetic changes to have a better global view

File: barman/postgres.py (hunk 2 above)
Comment (didiermichel) [reply barman/postgres.py]: Indeed, I though the username was the application name. 
changed it fo `self.conn_parameters.get("user")`

**Pull request merged**